### ISSUE TITLE
Optionally return a count for omQueryEntryPoints()

### ISSUE THREAD

**Issue by gongsu832** [onnx/onnx-mlir#1249]
@tungld I think we talked about the need for a count for the `omQueryEntryPoints()` call. Since most of the time, one typically needs to go through the list of entry points anyway, so a separate call to get the count is left out since one can always just iterate the NULL-terminated list to get the count.

As it turns out, there might be situations where we need to know the count **upfront**, even if we are going to go through the list later. As in the case of the JNI wrapper, the count needs to be known before we start iterating the entry points in order to create a Java String array object upfront to hold the entry points. Of course we can always iterate the list to get the count but iterating twice feels wasteful. So there is indeed a need for the count separately, even if we plan to iterate the NULL-terminated list.

But instead of creating another API call, it might be simpler to just augment `omQueryEntryPoints()` to take an "optional" `count` parameter, i.e.,
```
const char *const *omQueryEntryPoints(int *count);
```
Basically, if a non-NULL `count` is passed in, the count is returned. Regardless, the list is always NULL-terminated.

Comments/suggestions?

**Comment by gongsu832** [comment]
On a related issue, it appears that the current input/output signature functions have a bug that they return the last entry point's signatures when the input entry point doesn't exist.

**Comment by tungld** [comment]
@gongsu832 

> const char *const *omQueryEntryPoints(int *count);

I created a PR #1252 to support this function.

> On a related issue, it appears that the current input/output signature functions have a bug that they return the last entry point's signatures when the input entry point doesn't exist.

Though I have not yet checked this, I also modify the code to return NULL if the entry point does not exist.
